full-client/index.ts

```diff
@@ -1,34 +1,39 @@
 import 'source-map-support/register';
+import { Handler, ICoreConnectionOptions } from '@secret-agent/client';
+import Agent from '@secret-agent/client/lib/Agent';
+import CoreClient from '@secret-agent/client/lib/CoreClient';
+import ICoreRequestPayload from '@secret-agent/core-interfaces/ICoreRequestPayload';
+import CoreClientConnection from '@secret-agent/client/lib/CoreClientConnection';
 import Core from '@secret-agent/core';
-import { SecretAgentClientGenerator } from '@secret-agent/client';
-import ISessionMeta from '@secret-agent/core-interfaces/ISessionMeta';
+import CoreServerConnection from '@secret-agent/core/lib/CoreServerConnection';
 
-process.title = 'SecretAgent';
+export { Handler, Agent };
 
-const { SecretAgent, coreClient } = SecretAgentClientGenerator();
-
-// OUTGOING ////////////////////////////////////////////////////////////////////
-
-coreClient.pipeOutgoingCommand = async (
-  sessionMeta: ISessionMeta | null,
-  command: string,
-  args: any[],
-) => {
-  if (sessionMeta) {
-    const core = Core.byTabId[sessionMeta.tabId];
-    const data = await core[command](...args);
-    const commandId = core.lastCommandId;
-    return { data, commandId };
+class LocalCoreConnection extends CoreClientConnection {
+  constructor(
+    options: ICoreConnectionOptions,
+    readonly coreServerConnection: CoreServerConnection,
+  ) {
+    super(options);
+    coreServerConnection.on('message', payload => this.onMessage(payload));
   }
-  return { data: await Core[command](...args) };
-};
 
-// INCOMING ////////////////////////////////////////////////////////////////////
+  sendRequest(payload: ICoreRequestPayload): void | Promise<void> {
+    return this.coreServerConnection.handleRequest(payload);
+  }
+}
```

Comment:
This seems to want it's own file. lib/LocalCoreConnection.ts?

Reply:
Why not just put it in client/lib alongside RemoteCoreConnection.ts? The class itself doesn't need access to Core since it accepts coreServerConnection instance.

Reply:
CoreServerConnection is part of core, so it can't go in client, but we can definitely move it to it's own class. I had previously created a small interface that LocalCoreConnection took in, which I can do again, but I kind of liked seeing it wired up in the same file as the "LocalCoreConnectionCreator" part of the file so you could understand how it was getting wired up. I'm kind of cool with either path, but I tried several options here, and having it sit in the "middle or core and client" seemed to make the most sense - it feels like the many-to-many class.

Reply:
Your logic makes sense. 

I guess it just felt/feels weird to me that of our two CoreConnection classes, they're split across two different packages. It seems they should be together (along with UlixeeCoreConnection when we add that).

There's nothing else in full-client that defines new functionality or classes... it's just some dependencies and glue code.

Reply:
I guess I can see the appeal of having them near each other. However, they each have separate, optional dependencies. If we abstract the LocalCoreConnection so it takes in an interface, would you be in favor of a "connections" folder in client, so we don't have to flip the naming to the end to keep them in the same place? Or should I flip the names on all of them - CoreClientConnection, CoreConnectionRemote, CoreConnectionLocal, etc?

Reply:
Yes

Reply:
This has been updated, as well as unifying full-client and client exported classes/instances, and ability to set a CoreConnection on an agent